flexmeasures/data/models/planning/utils.py

```diff
@@ -28,20 +34,108 @@ def initialize_series(
     start: datetime,
     end: datetime,
     resolution: timedelta,
+    inclusive: str = "left",
 ) -> pd.Series:
-    s = pd.Series(index=initialize_index(start, end, resolution), data=data)
+    s = pd.Series(index=initialize_index(start, end, resolution, inclusive), data=data)
     return s
 
 
 def initialize_index(
-    start: Union[date, datetime], end: Union[date, datetime], resolution: timedelta
+    start: Union[date, datetime],
+    end: Union[date, datetime],
+    resolution: timedelta,
+    inclusive: str = "left",
 ) -> pd.DatetimeIndex:
+    assert inclusive == "left" or inclusive == "right"
```

Comment:
I don't think this is needed. There are other valid options, and Pandas already checks validity.